File: src/main/java/br/com/autoflex/service/RawMaterialService.java
package br.com.autoflex.service;

import br.com.autoflex.dto.RawMaterialDTO;
import java.util.List;

public interface RawMaterialService {
    RawMaterialDTO create(RawMaterialDTO dto);
    List<RawMaterialDTO> getAll();
    RawMaterialDTO getById(Long id);
    RawMaterialDTO update(Long id, RawMaterialDTO dto);
    void delete(Long id);
}

File: frontend-vue/src/services/api/rawMaterialService.ts
import { apiClient } from './config'
import type { RawMaterialDTO } from '@/types/api'

export const rawMaterialService = {
  async getAll(): Promise<RawMaterialDTO[]> {
    return apiClient.get<RawMaterialDTO[]>('/raw-materials')
  },
  async getById(id: number): Promise<RawMaterialDTO> {
    return apiClient.get<RawMaterialDTO>(`/raw-materials/${id}`)
  },
  async create(rawMaterial: Omit<RawMaterialDTO, 'id'>): Promise<RawMaterialDTO> {
    return apiClient.post<RawMaterialDTO>('/raw-materials', rawMaterial)
  },
  async update(id: number, rawMaterial: Omit<RawMaterialDTO, 'id'>): Promise<RawMaterialDTO> {
    return apiClient.put<RawMaterialDTO>(`/raw-materials/${id}`, rawMaterial)
  },
  async delete(id: number): Promise<void> {
    return apiClient.delete<void>(`/raw-materials/${id}`)
  },
}
